python/cuml/dask/ensemble/base.py

```diff
@@ -20,40 +38,32 @@ def _create_the_model(self, model_func, **kwargs):
 
         n_est_per_worker = math.floor(self.n_estimators / n_workers)
 
-        for i in range(n_workers):
-            self.n_estimators_per_worker.append(n_est_per_worker)
-
+        self.n_estimators_per_worker = \
+            [n_est_per_worker for i in range(n_workers)]

```

Comment:
So, this is like a default you're starting with and then you'll replace it with the true n_est_per_worker right? In that case, I would start it as a local and only assign it to self. after it is fully built. It is nice to ensure a general property that self. fields are coherent objects and never half-built.

Reply:
The `self.n_estimators_per_worker` here is the final one if `n_estimators` is divisible by `n_workers`.

Reply:
Could we wrap this logic in a function with a descriptive name? It would clean up this `create_models` code a little bit and make the intent more clear.

Reply:
@Salonijain27 yup, this field is sometimes valid and sometimes not (depending on divisibility). So we can just keep it as a local and then assign to the self field when it is ensured to be valid. Keeping self fields coherent and consistent goes a long way towards making code easier to reason about a codebase.

Reply:
i moved this to a function and now the output of the function is assigned to `self..n_estimators_per_worker `